Productionised UCM2 files

chadmed opened the pull request:
Tracking progress here, will rebase before merge

chadmed committed:
remove speaker detection

removed temporarily. will be added back later.

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
cs42l84: move headphone enable sequence to device

prevents overriding of SectionVerb

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
cs42l83: move adc boost switch to device enablesequence

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
laptops: explicitly set headphone playback mux

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
laptops: set target quality to hifi

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
restructure the entire overlay

since we now support more combinations of devices, descriminating
by only the jack codec is no longer feasible. restructure the directory
so that we can support more devices with the minimum extra files

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
laptops: add playback channels

this is required to get the speaker devices recognised correctly for
routing to the dsp graph

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
add conflictingdevice

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
headphones: add supporteddevice

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
cs42l84: make enable sequence more explicit

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
speakers: add enablesequence

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
remove conflictingdevice

doesnt work

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
pluralise speakers comment

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
all: detect paired jack codec automatically

this allows us to deduplicate the hifi files

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
all: reinstate speaker detection

required for smoothly transitioning speaker support on a per-machine
basis. on laptops, look for Left Woofer instead of Tweeter as some
four-speaker laptops do not have tweeters.

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
Fix new models

Update models added in 78db9b63f8 to use the new directory structure

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

jannau commented on ucm2/conf.d/macaudio/desktops/mono/HiFi.conf:
```diff
@@ -0,0 +1,59 @@
+If.cs42l84 {
+	Condition {
+		Type ControlExists
+		Control "name='Jack ADC WNF Switch'"
```
I would prefer some like
```
Define.JackCodec "cs42l84" 
```
in  `MacBook Pro J314.conf`
and
```
If.jackcodeccs42l84 {
	Condition {
		Type String
		String1 "cs42l84"
		String2 "${var:JackCodec}"
	}
	True.Include.hp4.File "/conf.d/macaudio/cs42l84/Headphones.conf"
}
```
here. same for cs42l83 and the other `Hifi.conf` files.

jannau commented on ucm2/conf.d/macaudio/laptops/4spk/HiFi.conf:
```diff
@@ -0,0 +1,56 @@
+If.havespk {
+	Condition {
+		Type ControlExists
+		Control "name='Left Woofer Speaker Volume'"
```
this doesn't work for all devices. j293 and j493 have "Left Front" and "Left Rear". The control for j293 is `Speaker Playback Volume`. I'd recommend using a define for this in the device `.conf` files

jannau commented on ucm2/conf.d/macaudio/desktops/mono/HiFi.conf:
```diff
@@ -0,0 +1,59 @@
+If.havespk {
+	Condition {
+		Type ControlExists
+		Control "name='Speaker Playback Volume'"
```
This should be `Speaker Volume` on all devices using tas2764

jannau commented on ucm2/conf.d/macaudio/desktops/mono/HiFi.conf:
```diff
@@ -0,0 +1,59 @@
+			Value {
+				PlaybackMixerElem "Speaker"
+				PlaybackPriority 100
+				PlaybackPCM "hw:${CardId},1"
```
unrelated to this merge request, it strikes me odd that we use for the jack sub device a define and hard-code 1 here. Would be more consistent to use `Define.SpeakerDevice "1"` and replace the 1 here with `${var:SpeakerDevice}`

jannau commented on ucm2/conf.d/macaudio/laptops/4spk/HiFi.conf:
```diff
@@ -0,0 +1,56 @@
+			]
+
+			Value {
+				PlaybackChannels 4
```
Can this be a variable? that would allow consolidating the 4spk and 6spk configs. We need a 2spk config for j313

chadmed committed:
rework jack detection

relying on an element name is gross, and we can't pull any
other distinguishing information out at runtime. just define
the jack codec used by each machine in its file.

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
laptops: define channels at the device level

this lets us remove another layer of duplication

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
move JackDevice and define SpeakerDevice

doing this at the device level lets us be more flexible with PCM
assignments in the future

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
define speaker ControlExists at device level

this lets us easily account for differences in how the driver
names speaker controls across devices

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

jannau commented on ucm2/conf.d/macaudio/MacBook Air J313.conf:
```diff
@@ -2,5 +2,10 @@ Syntax 6
 	Comment "Default"
-	File "laptops1/HiFi.conf"
+	Define.JackCodec "cs42l83"
+	Define.PlaybackChannels "4"
```
this should be `"2"`, j313 has only 2 speakers

jannau commented on ucm2/conf.d/macaudio/MacBook Air J313.conf:
```diff
@@ -2,5 +2,10 @@ Syntax 6
+	Define.PlaybackChannels "4"
+	Define.JackDevice "0"
+	Define.SpeakerDevice "1"
+	Define.SpeakerElem "Left Front"
```
should be `"Left Speaker Playback Volume"`

jannau commented on ucm2/conf.d/macaudio/MacBook Air J413.conf:
```diff
@@ -2,5 +2,10 @@ Syntax 6
+	Define.PlaybackChannels "4"
+	Define.JackDevice "0"
+	Define.SpeakerDevice "1"
+	Define.SpeakerElem "Left Front"
```
should be `"Left Woofer Speaker Volume"`

jannau commented on ucm2/conf.d/macaudio/MacBook Pro J293.conf:
```diff
@@ -2,5 +2,10 @@ Syntax 6
+	Define.PlaybackChannels "4"
+	Define.JackDevice "0"
+	Define.SpeakerDevice "1"
+	Define.SpeakerElem "Left Front"
```
should be `"Left Front Speaker Playback Volume"`

jannau commented on ucm2/conf.d/macaudio/MacBook Pro J493.conf:
```diff
@@ -2,5 +2,10 @@ Syntax 6
+	Define.PlaybackChannels "4"
+	Define.JackDevice "0"
+	Define.SpeakerDevice "1"
+	Define.SpeakerElem "Left Front"
```
should be `"Left Front Speaker Volume"`

jannau commented on ucm2/conf.d/macaudio/iMac J456.conf:
```diff
@@ -5,5 +5,10 @@ SectionUseCase."HiFi" {
 	Comment "Default"
-	File "laptops1/HiFi.conf"
+	Define.JackCodec "cs42l83"
+	Define.PlaybackChannels "4"
```
we have no speakers in out dts yet but imacs have 6 speakers with force cancelling woofers

jannau commented on ucm2/conf.d/macaudio/iMac J456.conf:
```diff
@@ -5,5 +5,10 @@ SectionUseCase."HiFi" {
+	Define.PlaybackChannels "4"
+	Define.JackDevice "0"
+	Define.SpeakerDevice "1"
+	Define.SpeakerElem "Left Front"
```
should be `"Left Woofer 1 Speaker Playback Volume"` if we use the same names in our dts as for the other devices.

jannau commented on ucm2/conf.d/macaudio/iMac J457.conf:
```diff
@@ -5,5 +5,10 @@ SectionUseCase."HiFi" {
+	Define.PlaybackChannels "4"
+	Define.JackDevice "0"
+	Define.SpeakerDevice "1"
+	Define.SpeakerElem "Left Front"
```
see comment for j456

chadmed committed:
fix device specific data

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
codecs: remove SupportedDevice

this was causing the UCM parsing code to become confused and skip
device enablement

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
single-quote enum values

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
cs42l84: set some sane WNF and HPF defaults

this jack cant be used as a line in anyway, so at least make peoples'
mics sound alright out of the box

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
cs42l84: hook up physical ADC volume control

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
cs42l84: set mic preamp gain to +20 dB

produces pleasing volume scaling

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>

chadmed committed:
cs42l84: crank up the preamp gain

the PGA volume introduces hiss at the top end. crank up
the preamp gain to get better snr out of the ADC

Signed-off-by: James Calligeros <jcalligeros99@gmail.com>
